documentation/src/docs/asciidoc/user-guide/extensions.adoc

```diff
@@ -359,6 +359,16 @@ initialization methods on the test instance, etc.
 For a concrete example, consult the source code for the `{MockitoExtension}` and the
 `{SpringExtension}`.
 
+[[extensions-test-instance-pre-construct-callback]]
+=== Test Instance Pre-construct Callback
```

Comment:
I wonder if this section should go above `TestInstanceFactory` since it comes before in the call sequence. Thoughts?

Reply:
I agree - I thought about it too when I was writing it, makes sense.